syntax = "proto2";

option optimize_for = LITE_RUNTIME;

package authpolicy.protos;

// Active directory information entered during domain join.
message ActiveDirectoryConfig {
  optional string machine_name = 1;
  optional string realm = 3;
}

// Information about a GPO parsed from net ads gpo list. |name| is the 16-byte
// guid (e.g. {12345678-90AB-...}). |basepath| and |directory| are created from
// the first and the last part of the GPO's 'filesyspath' value in the net
// output, e.g. if filesyspath is
// \\example.com\SysVol\example.com\Policies\{12345678-90AB-CDEF-1234-567890ABCDEF},
// then |basepath| is 'example.com/SysVol' and |directory| is
// example.com\Policies\{12345678-90AB-CDEF-1234-567890ABCDEF}.
message GpoEntry {
  optional string name = 1;
  optional string basepath = 2;
  optional string directory = 3;
}

// List of GPOs on server. Agnostic of policy scope (user/machine); GPOs can
// contain both user and machine policy.
message GpoList {
  repeated GpoEntry entries = 1;
}

// Local file paths of downloaded GPO files.
message FilePathList {
  repeated string entries = 1;
}

// Account details obtained from "net ads search ...".
message AccountInfo {
  // The unique identifier for an Active Directory object. |objectGUID|
  // attribute is used.
  optional string object_guid = 1;
  // The logon name. |sAMAccountName| attribute is used.
  optional string sam_account_name = 2;
  // Display name of the user. |displayName| attribute is used.
  optional string display_name = 3;
  // Given name of the user. AKA first name. |givenName| attribute is used.
  optional string given_name = 4;
}

// Selected data from net ads info. |dc_name| is the name of the domain
// controller and |kdc_ip| is the IPv4 or IPv6 address of the key distribution
// center.
message RealmInfo {
  optional string dc_name = 1;
  optional string kdc_ip = 2;
}
